Generals/Code/GameEngine/Source/GameClient/InGameUI.cpp

```diff
@@ -5428,21 +5428,33 @@ void InGameUI::selectNextIdleWorker( void )
 	else
 	{
 		Drawable *selectedDrawable = TheInGameUI->getFirstSelectedDrawable();	
+		ObjectList uniqueIdleWorkers;
+
+		// Find unique selectables to avoid selecting the same or a previous container if multiple idle workers are contained
+		for (ObjectListIt it = m_idleWorkers[index].begin(); it != m_idleWorkers[index].end(); ++it)
```

Comment:
This will be better in a new function with a good name.

`static std::vector<Object*> buildUniqueIdleWorkers(const ObjectList& idleWorkers)`